api/v1beta1/agent_types.go

```diff
@@ -207,11 +207,13 @@ type AgentStatus struct {
 	Hostname  string `json:"hostname,omitempty"`
 	Bootstrap bool   `json:"bootstrap,omitempty"`
 	// +optional
-	Role       models.HostRole          `json:"role" protobuf:"bytes,1,opt,name=role,casttype=HostRole,omitempty"`
-	Inventory  HostInventory            `json:"inventory,omitempty"`
-	Progress   HostProgressInfo         `json:"progress,omitempty"`
-	NtpSources []HostNTPSources         `json:"ntpSources,omitempty"`
-	Conditions []conditionsv1.Condition `json:"conditions,omitempty"`
+	Role models.HostRole `json:"role" protobuf:"bytes,1,opt,name=role,casttype=HostRole,omitempty"`
+	// +optional
+	SuggestedRole models.HostRole          `json:"suggested_role" protobuf:"bytes,1,opt,name=suggested_role,casttype=HostRole,omitempty"`
```

Comment:
@slaviered @filanov Do we want to expose SuggestedRole in the k8s API, or combine it with Role?

Reply:
In 2.5 we probably should, @slaviered is there a ticket for that? 
In 2.4 i think that this feature should be disabled, i'm not confident in how mature it is, @slaviered what do you think?

Reply:
I am disabling the main functionality of the feature with a feature flag that prevents the validations to use the suggested role as an effective role for their decision algorithm. Here it is just a field in the status that exposes the suggested role and it has no functional effect

Reply:
Per our discussion, only Role field will be present in the Status